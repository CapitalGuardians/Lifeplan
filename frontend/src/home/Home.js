import React from "react";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import Grid from "@material-ui/core/Grid";
import CardContent from "@material-ui/core/CardContent";
import { makeStyles } from "@material-ui/styles";
import { DARK_BLUE, LIGHT_BLUE, NAV_BAR_HEIGHT } from "../common/theme";
import Typography from "@material-ui/core/Typography";
import classNames from "classnames";

const useStyles = makeStyles({
  home: {
    height: `calc(100vh - ${NAV_BAR_HEIGHT}px)`
  },
  planPageSelector: {
    backgroundColor: DARK_BLUE,
    color: "white"
  },
  budgetPageSelector: {
    background: LIGHT_BLUE,
    color: DARK_BLUE
  },
  pageSelector: {
    width: 140,
    height: 140,
    textAlign: "center",
    display: "flex",
    alignItems: "center"
  },
  cardContainer: {
    marginTop: 40,
    marginLeft: "10vw",
    marginRight: "10vw"
  }
});

export default function Home() {
  const classes = useStyles();
  return (
    <Grid container alignItems="center" className={classes.home}>
      <Grid item xs={12}>
        <Grid container>
          <Grid item xs={12}>
            <Typography variant="h5" align="center">
              Welcome.
              <br />
              Please select the tool you wish to use:
            </Typography>
          </Grid>

          <Grid item xs={12}>
            <Grid container justify="center">
              <Grid item>
                <Card raised className={classes.cardContainer}>
                  <CardActionArea
                    className={classNames(
                      classes.planPageSelector,
                      classes.pageSelector
                    )}
                  >
                    <CardContent>
                      <Typography variant="h6" align="center" color="inherit">
                        Plan Creator
                      </Typography>
                    </CardContent>
                  </CardActionArea>
                </Card>
              </Grid>

              <Grid item>
                <Card
                  raised
                  className={classes.cardContainer}
                  onClick={() => (window.location.href = "/budget")}
                >
                  <CardActionArea
                    className={classNames(
                      classes.budgetPageSelector,
                      classes.pageSelector
                    )}
                  >
                    <CardContent>
                      <Typography variant="h6" align="center" color="inherit">
                        Plan Budget Calculator
                      </Typography>
                    </CardContent>
                  </CardActionArea>
                </Card>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
}
