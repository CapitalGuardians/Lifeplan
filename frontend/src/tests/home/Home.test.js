import React from "react";
import Home from "../../home/Home";
import renderer from "react-test-renderer";

test("Test home rendering", () => {
  const component = renderer.create(<Home />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
