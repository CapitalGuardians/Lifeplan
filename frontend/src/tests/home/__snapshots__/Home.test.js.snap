// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test home rendering 1`] = `
<div
  className="MuiGrid-root makeStyles-home-1 MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
            >
              Welcome.
              <br />
              Please select the tool you wish to use:
            </h5>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiPaper-root MuiPaper-elevation8 MuiPaper-rounded MuiCard-root"
                >
                  <button
                    className="MuiButtonBase-root MuiCardActionArea-root makeStyles-planPageSelector-2 makeStyles-pageSelector-4"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      className="MuiCardContent-root"
                    >
                      <h6
                        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-alignCenter"
                      >
                        Plan Creator
                      </h6>
                    </div>
                    <span
                      className="MuiCardActionArea-focusHighlight"
                    />
                  </button>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiPaper-root MuiPaper-elevation8 MuiPaper-rounded MuiCard-root"
                  onClick={[Function]}
                >
                  <button
                    className="MuiButtonBase-root MuiCardActionArea-root makeStyles-budgetPageSelector-3 makeStyles-pageSelector-4"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      className="MuiCardContent-root"
                    >
                      <h6
                        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-alignCenter"
                      >
                        Plan Budget Calculator
                      </h6>
                    </div>
                    <span
                      className="MuiCardActionArea-focusHighlight"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
